bedrock/mozorg/tests/test_helper_misc.py

```diff
@@ -863,6 +863,37 @@ def test_pocket_play_store_url_localized_campaign(self):
         )
 
 
+class TestMSStoreURL(TestCase):
+    rf = RequestFactory()
+
+    def _render(self, product, campaign, mode):
+        req = self.rf.get("/")
+        return render(
+            f"{{{{ ms_store_url('{product}', '{campaign}', '{mode}') }}}}",
+            {"request": req},
+        )
+
+    def test_firefox_release_ms_store_url(self):
+        """should return a MS Store URL for Firefox release channel"""
+        assert self._render("firefox", "", "direct") == "https://apps.microsoft.com/detail/9nzvdkpmr9rd?mode=direct"
+
+    def test_firefox_beta_ms_store_url(self):
+        """should return a MS Store URL for Firefox Beta channel"""
+        assert self._render("firefox_beta", "", "direct") == "https://apps.microsoft.com/detail/9nzw26frndln?mode=direct"
+
+    def test_firefox_ms_store_url_launch_mode(self):
+        """should return a MS Store URL including different launch mode parameters"""
+        assert self._render("firefox", "", "full") == "https://apps.microsoft.com/detail/9nzvdkpmr9rd?mode=full"
+        assert self._render("firefox", "", "mini") == "https://apps.microsoft.com/detail/9nzvdkpmr9rd?mode=mini"
+
+    def test_firefox_ms_store_url_campaign(self):
+        """should return a MS Store URL including campaign parameters"""
+        assert (
+            self._render("firefox", "mozorg-firefox-home", "direct")
+            == "https://apps.microsoft.com/detail/9nzvdkpmr9rd?mode=direct&amp;cid=mozorg-firefox-home"
```

Comment:
Is the `&amp;` correct there? I would expect that to just be `&` for the additional query parameters.

Reply:
I don't think `&amp;` is technically needed when it's a query param separator, but it doesn't really cause any issues since browsers will still interpret them correctly as links. This seems to be Django's default behavior when rendering URL strings like this in templates, unless you explicitly mark it as safe.